hrm-domain/hrm-service/service-tests/contact-job/jobTypes/scrubTranscripts.test.ts

```diff
@@ -278,4 +291,91 @@ describe('Scrub job complete', () => {
     );
     expect(await pullDueContactJobs(new Date(), 5)).toHaveLength(0);
   });
+
+  test('Receive a completed retrieve transcript job and create a scrub job', async () => {
+    await contactApi.addConversationMediaToContact(
+      accountSid,
+      testContactId.toString(),
+      [
+        {
+          storeType: 'S3',
+          storeTypeSpecificData: {
+            type: S3ContactMediaType.SCRUBBED_TRANSCRIPT,
+            location: { bucket: 'mock-bucket', key: 'mock-new-scrubbed-transcript-path' },
+          },
+        },
+      ],
+      ALWAYS_CAN,
+      true,
+    );
+
+    await createContactJob()({
+      resource: await contactApi.getContactById(accountSid, testContactId, ALWAYS_CAN),
+      jobType: ContactJobType.SCRUB_CONTACT_TRANSCRIPT,
+      additionalPayload: {
+        originalLocation: {
+          bucket: 'mock-bucket',
+          key: 'mock-transcript-path',
+        },
+      },
+    });
+    const [pendingScrubJobs] = await pullDueContactJobs(new Date(), 5);
+    const message = {
+      jobId: pendingScrubJobs.id,
+      jobType: ContactJobType.RETRIEVE_CONTACT_TRANSCRIPT,
+      originalLocation: { bucket: 'mock-bucket', key: 'mock-transcript-path' },
+      taskId: 'TKx',
+      twilioWorkerId: workerSid,
+      contactId: testContactId,
+      accountSid,
+      attemptNumber: 0,
+      attemptPayload: {
+        retrievedLocation: {
+          bucket: 'mock-bucket',
+          key: 'mock-retrieved-transcript-path',
+        },
+      },
+      attemptResult: ContactJobAttemptResult.SUCCESS,
+    };
+
+    await sqsClient
+      .sendMessage({
+        QueueUrl: completedQueueUrl,
+        MessageBody: JSON.stringify(message),
+      })
+      .promise();
+
+    await singleProcessContactJobsRun();
+
+    const updatedContact = await contactApi.getContactById(
+      accountSid,
+      testContactId,
+      ALWAYS_CAN,
+    );
+
+    expect(updatedContact.conversationMedia?.length).toBe(2);
+
+    verifyConversationMedia(
+      updatedContact,
+      S3ContactMediaType.TRANSCRIPT,
+      'mock-transcript-path',
+    );
+    verifyConversationMedia(
+      updatedContact,
+      S3ContactMediaType.SCRUBBED_TRANSCRIPT,
+      'mock-new-scrubbed-transcript-path',
+    );
+
+    expect(pendingScrubJobs).toBeDefined();
+    expect(pendingScrubJobs.jobType).toBe(ContactJobType.SCRUB_CONTACT_TRANSCRIPT);
+
+    const messages = await sqsClient
+      .receiveMessage({
+        QueueUrl: pendingScrubJobsQueueUrl,
+        MaxNumberOfMessages: 1,
+      })
+      .promise();
+
+    expect(messages).toBeDefined();
```

Comment:
`messages` will always be defined because it is an array. You need to check if there is a message in the array, and validate the contents are as expected.

Reply:
@stephenhand - `messages` is not an array but an object that contains metadata.